pkg/config/remote/api/http_test.go

```diff
@@ -0,0 +1,166 @@
+package api
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config/mock"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test WebSocket connectivity under varying transport configurations, and
+// assert various transport-level properties such as credential transmission.
+func TestNewWebSocket(t *testing.T) {
+	upgrader := websocket.Upgrader{}
+	auth := Auth{
+		APIKey:    "bananas",
+		PARJWT:    "platanos",
+		AppKey:    "banana_key",
+		UseAppKey: true,
+	}
+
+	testCases := []struct {
+		desc string
+		path string
+
+		serverHTTP2 bool
+		serverTLS   bool
+	}{
+		{
+			desc:        "tls with http1",
+			path:        "/",
+			serverHTTP2: false,
+			serverTLS:   true,
+		},
+		{
+			desc:        "tls with http2",
+			path:        "/",
+			serverHTTP2: true,
+			serverTLS:   true,
+		},
+		{
+			desc:        "plain with http1",
+			path:        "/",
+			serverHTTP2: false,
+			serverTLS:   false,
+		},
+		{
+			desc:        "plain with http2",
+			path:        "/",
+			serverHTTP2: true,
+			serverTLS:   false,
+		},
+		{
+			desc:        "path",
+			path:        "/api/v0.2/echo",
+			serverHTTP2: false,
+			serverTLS:   false,
+		},
```

Comment:
Varying transport configurations are tested:

* TLS on/off
* HTTP vs HTTP2

This is probably going to be relevant in deployments that use non-transparent HTTP egress proxies irrespective of what RC uses in the backend.